sycl/include/sycl/handler.hpp

```diff
@@ -2150,6 +2191,50 @@ class __SYCL_EXPORT handler {
     parallel_for_impl<KernelName>(Range, Properties, std::move(KernelFunc));
   }
 
+  template <typename KernelName, typename KernelType, int Dims,
```

Comment:
I don't see `KernelName` being used here at all. Should we propagate it (in some probably modified form similar to range rounding code above) down to wrapper kernels that we launch?